pkg/controllers/dynakube/activegate/reconciler.go

```diff
@@ -97,24 +99,29 @@ func (r *Reconciler) Reconcile(ctx context.Context) error {
 		}
 		// TODO: have a cleanup for things that we create above
 
-		err := r.authTokenReconciler.Reconcile(ctx)
+		err = r.authTokenReconciler.Reconcile(ctx)
 		if err != nil {
 			return errors.WithMessage(err, "could not reconcile Dynatrace ActiveGateAuthToken secrets")
 		}
+	} else {
+		statusconditions := &r.dynakube.Status.Conditions
+		meta.RemoveStatusCondition(statusconditions, conditions.ActiveGateConnectionInfoConditionType)
+		meta.RemoveStatusCondition(statusconditions, conditions.ActiveGateStatefulSetConditionType)
+		meta.RemoveStatusCondition(statusconditions, conditions.ActiveGateVersionConditionType)
```

Comment:
each sub reconciler should handle its own conditions, so we can avoid this

Reply:
Problem is that the sub reconcilers are not called if activegate is removed from the dynakube. Therefore they would not run and cleanup the conditions